### ISSUE TITLE
Self-defined C/C++ toolchain with "no_legacy_features" does not accept unknown features

### ISSUE THREAD

**Issue by andreas-0815-qwertz** [bazelbuild/bazel#23188]
### Discussed in https://github.com/bazelbuild/bazel/discussions/22992

<div type='discussions-op-text'>

<sup>Originally posted by **andreas-0815-qwertz** July 11, 2024</sup>
I am creating a toolchain for Gcc 3.4.5; this Gcc version does not support parameter files for the linker (`linker_param_file` feature). Therefore, I have defined the feature "no_legacy_features" in my toolchain. Compiler and archiver features are supported by issuing the related flags in the command lines, but the linker features are unfortunately not.  I am sure I am missing something but could you please tell me what?

I have attached a minimal working example using my toolchain with `/usr/bin/gcc`. On a linux system you should be able to unpack the tarball and run `bazel build --noenable_bzlmod --platforms=//bazel_utils/platforms:onepc_platform -s //:toolchain_test` to see, what I mean. The surprising thing for me is that the feature `default_link_flags` does not seem to work.  `/usr/bin/gcc` is called as the linker but without any command line arguments, so that the build fails.

My bazel configuration is:

$ bazel version
Bazelisk version: v1.17.0
Build label: 7.2.1
Build target: @@//src/main/java/com/google/devtools/build/lib/bazel:BazelServer
Build time: Tue Jun 25 15:53:05 2024 (1719330785)
Build timestamp: 1719330785
Build timestamp as int: 1719330785

Edit: Fixed feature name and added short description of the actual error.

[toolchain_test.tar.gz](https://github.com/user-attachments/files/16172429/toolchain_test.tar.gz)
</div>

**Comment by pzembrod** [comment]
Hi,

IIUC the question was solved in https://stackoverflow.com/questions/73504780/bazel-reference-binaries-from-packages-in-custom-toolchain-definition which is linked in your code example. Correct?

**Comment by andreas-0815-qwertz** [comment]
No, I am afraid the link that I have added to the file `cc_gcc345_linux_cross_config.bzl` is unrelated to the above problem.  If you run the build as suggested, you will find that `/usr/bin/gcc` is called as the linker but without any command line flags.  This is despite the fact that the linker feature `default_link_flags` is enabled by default.  I am a bit puzzled, why that is.

**Comment by pzembrod** [comment]
Actually, when I try to repro with `USE_BAZEL_VERSION=7.2.1 bazelisk  build --noenable_bzlmod --platforms=//bazel_utils/platforms:onepc_platform -s //:toolchain_test`, /usr/bin/gcc gets called without any params and fails.

At the moment I'm afraid I can't give any advice why this might be.
